src/Features/Core/Portable/CodeRefactorings/UseNamedArguments/AbstractUseNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -19,17 +19,22 @@ protected interface IAnalyzer
                 CodeRefactoringContext context, SyntaxNode root, CancellationToken cancellationToken);
         }
 
-        protected abstract class Analyzer<TBaseSyntax, TSyntax, TListSyntax> : IAnalyzer
-            where TBaseSyntax : SyntaxNode
-            where TSyntax : TBaseSyntax
-            where TListSyntax : SyntaxNode
+        protected abstract class Analyzer<TBaseArgumentSyntax, TArgumentSyntax, TArgumentListSyntax> : IAnalyzer
+            where TBaseArgumentSyntax : SyntaxNode
+            where TArgumentSyntax : TBaseArgumentSyntax
+            where TArgumentListSyntax : SyntaxNode
         {
             public async Task ComputeRefactoringsAsync(
                 CodeRefactoringContext context, SyntaxNode root, CancellationToken cancellationToken)
             {
                 var document = context.Document;
 
-                var argument = root.FindNode(context.Span).FirstAncestorOrSelf<TBaseSyntax>() as TSyntax;
+                if (context.Span.Length > 0)
```

Comment:
:memo: I didn't see any new tests for non-empty selections. What was the old behavior?